compute_sdk/tests/unit/test_client.py

```diff
@@ -419,3 +429,35 @@ def test_version_mismatch_from_details(mocker, gcc, mock_response, dill_python):
             assert f"but worker used {worker_python}" in a[0]
         if worker_dill:
             assert f"/{worker_dill} (OS" in a[0]
+
+
+@pytest.mark.parametrize("num_runs", range(2, 10))
+def test_client_only_warns_version_mismatch_once(mocker, gcc, mock_response, num_runs):
+    tid = str(uuid.uuid4())
+    returned_task = {
+        "task_id": tid,
+        "status": "success",
+        "completion_t": "1677183605.212898",
+        "exception": "Some Stack Trace\n  Line 40 stack trace",
+        "details": {
+            "os": "Linux-5.19.0-1025-aws-x86_64-with-glibc2.35",
+            "python_version": "python-bad-version",
+            "dill_version": "dill-bad-version",
+            "globus_compute_sdk_version": "2.3.2",
+            "task_transitions": {
+                "execution-start": 1692742841.843334,
+                "execution-end": 1692742846.123456,
+            },
+        },
+    }
+
+    mock_log = mocker.patch("globus_compute_sdk.sdk.client.logger")
+
+    gcc.web_client.get_task.return_value = mock_response(200, returned_task)
+
+    for i in range(num_runs):
+        with pytest.raises(TaskExecutionFailed) as exc_info:
+            gcc.get_task(tid)
+        assert "Some Stack Trace" in str(exc_info), f"run {i} had unexpected exception"
+
+    mock_log.warning.assert_called_once()
```

Comment:
This puts in mind to ask the question: have we thought about what to do if the user uses multiple endpoints?  For example, what should we do in the following interaction?
```python
with Executor() as ex:
    ex.endpoint_id = "ep_id1"
    ex.submit(...).result()
    ex.endpoint_id = "ep_id2"
    ex.submit(...).result()
```

Reply:
Per Slack discussion, suggestion is to reset the flag when the `endpoint_id` is reset.  If the user gets multiple logs at that time, then ... "don't change the `endpoint_id` so often!"

Reply:
I ended up changing the behavior to store a `set` of already warned EP IDs on the Client, to handle cases like these.